projects/custody-key-dates-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/messaging/Handler.kt

```diff
@@ -24,11 +24,18 @@ class Handler(
                 ?.let { cduService.updateCustodyKeyDates(it) }
 
             is CustodyDateChanged -> cduService.updateCustodyKeyDates(message.bookingId)
+            is ProbationOffenderEvent -> when (notification.eventType) {
+                "SENTENCE_CHANGED",
+                -> cduService.updateCustodyKeyDates(cduService.findNomsByCrn(message.crn))
```

Comment:
So if you did turn the NOMS number lookup into a JPA query with a nullable response, you could change this to something like:
```suggestion
                -> cduService.findNomsByCrn(message.crn)?.let { cduService.updateCustodyKeyDates(it) }
```

Reply:
done